connect_ext_test.go

```diff
@@ -699,6 +736,43 @@ func TestBidiRequiresHTTP2(t *testing.T) {
 	)
 }
 
+func TestCompressMinBytesClient(t *testing.T) {
+	t.Parallel()
+	serve := func(request *pingv1.PingRequest) (string, error) {
+		mux := http.NewServeMux()
+		var contentEncoding string
+		go mux.Handle("/", http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
+			contentEncoding = request.Header.Get("Content-Encoding")
+		}),
+		)
+		server := httptest.NewServer(mux)
+		t.Cleanup(server.Close)
+		pingclient := pingv1connect.NewPingServiceClient(
+			http.DefaultClient, server.URL,
+			connect.WithSendGzip(),
+			connect.WithCompressMinBytes(8),
+		)
+		_, err := pingclient.Ping(context.Background(), connect.NewRequest(request))
+		return contentEncoding, err
+	}
```

Comment:
Love that we're testing this! Even with the updates, though, the approach seems a little convoluted. I think we can simplify a bit by having the helper function take the expected encoding as an argument and make the assertions:

```go
func assertContentType = func(tb testing.TB, text, expect string) {
  tb.Helper()
  mux := http.NewServeMux()
  mux.Handle("/", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
    assert.Equal(tb, request.Header.Get("Content-Encoding"), expect)
  })
  server := httptest.NewServer(mux)
  tb.Cleanup(server.Close)
  _, err := pingv1connect.NewPingServiceClient(
    server.Client,
    server.URL,
    connect.WithSendGzip(),
    connect.WithCompressMinBytes(8),
  ).Ping(context.Background(), connect.NewRequest(&pingv1.PingRequest{Text: text})
  assert.Nil(tb, err)
}
t.Run("request_uncompressed", func(t *testing.T) {
    assertContentType(t, "ping", "")
})
t.Run("request_compressed", func(t *testing.T) {
    assertContentType(t, "pingping", "gzip")
})
```

Reply:
Yeah this makes a lot more sense than what's there